test/utils/asserts.hpp

```diff
@@ -96,7 +97,6 @@ namespace fizzy::test
 {
 inline uint32_t as_uint32(fizzy::Value value)
```

Comment:
Eventually we should also remove this, if we merge this PR.

Reply:
@chfast do you want to remove it in a separate commit, or a separate PR? I can do the removal too.

Reply:
Well, actually shouldn't all the test code be using `as<uint32_t>()`? We can apply that parallel to this PR.

Reply:
> Well, actually shouldn't all the test code be using `as<uint32_t>()`? We can apply that parallel to this PR.

The `Result()` was checking this to some distinct. But I went with additional type checking to solved it for good.

Reply:
> @chfast do you want to remove it in a separate commit, or a separate PR? I can do the removal too.

In https://github.com/wasmx/fizzy/pull/691.